src/Compilers/CSharp/Test/Semantic/Semantics/UninitializedNonNullableFieldTests.cs

```diff
@@ -244,44 +247,157 @@ public void StaticFields_DefaultConstructor()
-            comp.VerifyDiagnostics();
-            //// (8,12): warning CS8618: Non-nullable field 'F3' is uninitialized. Consider declaring the field as nullable.
-            ////     static C()
-            //Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "C").WithArguments("field", "F3").WithLocation(8, 12),
-            //// (8,12): warning CS8618: Non-nullable field 'F1' is uninitialized. Consider declaring the field as nullable.
-            ////     static C()
-            //Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "C").WithArguments("field", "F1").WithLocation(8, 12));
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (16,12): warning CS8618: Non-nullable field 'F4' is uninitialized. Consider declaring the field as nullable.
+                //     static C()
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "C").WithArguments("field", "F4").WithLocation(16, 12),
+                // (16,12): warning CS8618: Non-nullable property 'P1' is uninitialized. Consider declaring the property as nullable.
+                //     static C()
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "C").WithArguments("property", "P1").WithLocation(16, 12),
+                // (16,12): warning CS8618: Non-nullable property 'P4' is uninitialized. Consider declaring the property as nullable.
+                //     static C()
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "C").WithArguments("property", "P4").WithLocation(16, 12),
+                // (16,12): warning CS8618: Non-nullable field 'F1' is uninitialized. Consider declaring the field as nullable.
+                //     static C()
+                Diagnostic(ErrorCode.WRN_UninitializedNonNullableField, "C").WithArguments("field", "F1").WithLocation(16, 12));
+        }
+
+        [Fact]
+        [WorkItem(34668, "https://github.com/dotnet/roslyn/issues/34668")]
+        public void StaticFields_GenericTypes()
+        {
+            var source =
+@"#pragma warning disable 0169
+class C<T, U, V>
+    where U : class
+    where V : struct
+{
+    private static T P1 { get; }
+    private static T P2 { get; } = default!;
+    private static U P3 { get; set; }
+    private static U P4 { get; set; } = default!;
+    private static U? P5 { get; }
+    private static U? P6 { get; } = default;
+    private static V P7 { get; set; }
+    private static V P8 { get; set; } = default;
+    private static V? P9 { get; }
+    private static V? P10 { get; } = default;
+}";
+            var comp = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp.VerifyDiagnostics(
+                // (6,22): warning CS8618: Non-nullable property 'P1' is uninitialized. Consider declaring the property as nullable.
```

Comment:
>Consider declaring the property as nullable. [](start = 88, length = 44)

This is bad. You can't declare that property as nullable.

Reply:
Marking the member as `[MaybeNull]` should suppress the warning. But it does not currently. Logged [https://github.com/dotnet/roslyn/issues/37511](https://github.com/dotnet/roslyn/issues/37511)

---
In reply to: [307842392](https://github.com/dotnet/roslyn/pull/37507#discussion_r307842392) [](ancestors = 307842392)

Reply:
Is this behavior consistent with instance fields of unconstrained generic types?

Reply:
Yes.

---
In reply to: [307930681](https://github.com/dotnet/roslyn/pull/37507#discussion_r307930681) [](ancestors = 307930681)